modules/http/http-loadbalancer.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2018 Balazs Scheidler
+ * Copyright (c) 2018 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef HTTP_LOADBALANCER_H_INCLUDED
+#define HTTP_LOADBALANCER_H_INCLUDED 1
+
+#include "syslog-ng.h"
+
+typedef enum
+{
+  HTTP_TARGET_OPERATIONAL,
+  HTTP_TARGET_FAILED
+} HTTPLoadBalancerTargetState;
+
+typedef struct _HTTPLoadBalancerTarget HTTPLoadBalancerTarget;
+typedef struct _HTTPLoadBalancerClient HTTPLoadBalancerClient;
+typedef struct _HTTPLoadBalancer HTTPLoadBalancer;
+
+
+/* NOTE: this struct represents an actual HTTP target URL.  The existance of
+ * this structure is ensured even in multi-threaded environments.  Some of
+ * the members of the struct are read-only and are _always_ available
+ * without locking.  Others are protected by the LoadBalancer's lock and
+ * should be manipulated under the protection of that lock.
+ */
+struct _HTTPLoadBalancerTarget
+{
+  /* read-only data, no need to lock */
+  gchar *url;
+  gint index;
+  /* read-write data, locking must be in effect */
+  HTTPLoadBalancerTargetState state;
+  gint number_of_clients;
+  gint max_clients;
+  time_t last_failure_time;
+};
+
+struct _HTTPLoadBalancerClient
```

Comment:
Note1:
We've talked about this IRL, but for completeness I'd still mention that I have the feeling that this hurts the YAGNI principle... on the other side, I know that when we extend the functionality, we will definitely need this structure.
Note 2: 
Maybe I'd use an index (that points to the targets array) instead of storing a pointer, but it's just a matter of taste (index or pointer... the result is almost the same, for me it is easier to read the code when index is used).